packages/txservice/src/provider.ts

```diff
@@ -113,71 +108,32 @@ export class ChainRpcProvider {
 
     // TODO: #146 We may ought to do this instantiation in the txservice constructor.
     this.signer = typeof signer === "string" ? new Wallet(signer, this.provider) : signer.connect(this.provider);
+
+    // TODO: This is an antipattern, or at least feels like one, in conjuction with the
+    // createTransaction() and sendTransaction() methods below. If sendTransaction() should only ever be called within
+    // Transaction class's submit() method, why should it be exposed below?
```

Comment:
good q

Reply:
yep, this is the last "antipattern" related q, and it's kind of a biggie. There's got to be a more elegant way to structure this, but for now, if it works, it works.

Reply:
Can you explain more about why it *is* exposed after posing the question/describe the antipattern you're concerned about in a bit more depth?